### PR TITLE
[Proposal] Create quest_status, quest_log and fame_area enum files

### PR DESCRIPTION
**Pull request by Xaver-DaRed**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

- Creates 3 enum files: quest_status, quest_log and fame_area enum files in the enum folder.
- Repace all instances of QUEST_AVAILABLE, QUEST_ACCEPTED and QUEST_COMPLETED to use the newly created enum.
- Replaces all instances of the old enum to the new one.
- Most importantly, remove QUEST_AVAILABLE, QUEST_ACCEPTED and QUEST_COMPLETED from the lua CI check exceptions

## Steps to test these changes

Review by commit.
- The first commit contains the important changes.
- The remaining commits contain the mass search/replaces

### CODE REVIEW

**Comment by Xaver-DaRed** [comment]
Updated proposal based on feedback

**Comment by zach2good** [comment]
I don't have any horses in this race, but it looks fine to me - I'll leave it up to the others to approve if they like it

**Review by claywar — APPROVED** [review]
Good with this, just needs a rebase (sorry), and bring the pain

**Pull request merged**
